### PR TITLE
TMDM-10011 Restrict journal view with role

### PR DESCRIPTION
**Pull request by yjli-talend**
https://jira.talendforge.org/browse/TMDM-10011
**What is the current behavior?** (You should also link to an open issue here)
The journal button of record detail toolbar will open 'Journal' function,we can see all 'journal' records in this function.

What is the new behavior?
The journal button of record detail toolbar will open 'User Journal' function, The User Journal will be only for single record and the pagination for the record's journal. We only see the records in view we have permission. The view and key search condition cannot be blank.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages.properties [hunk 1]
```diff
@@ -48,4 +48,7 @@ matching_failed=Entity {0} has no match rule or is not applicable to matching pr
 delete_success_prefix=Deletion succeeded message: 
 delete_fail_prefix=Deletion failed message: 
 restore_success_prefix=Restoration succeeded message: 
-restore_fail_prefix=Restoration failed message: 
+restore_fail_prefix=Restoration failed message:
+empty_entity = Select an Entity...
+empty_view = Select a View...
+loading =Loading...
```
[the remark this answers is not in the document]
Comment (pbabin) [reply org.talend.mdm.webapp.base/src/main/resources/org/talend/mdm/webapp/base/client/i18n/BaseMessages.properties]: Also we should remove the  ^M linebreak characters

File: org.talend.mdm.webapp.browserecords/src/main/resources/org/talend/mdm/webapp/browserecords/client/i18n/BrowseRecordsMessages.properties [hunk 2]
```diff
@@ -25,6 +25,7 @@ duplicate_btn = Duplicate
 duplicate_tip = Clone the selected record
 journal_btn = Journal
 journal_tip = Browse the audit trail for this record
+userJournal_btn = User Journal
```
Comment (pbabin) [inline org.talend.mdm.webapp.browserecords/src/main/resources/org/talend/mdm/webapp/browserecords/client/i18n/BrowseRecordsMessages.properties]: We should remove the  ^M linebreak characters

File: org.talend.mdm.webapp.general/src/main/java/org/talend/mdm/webapp/general/server/util/Utils.java [hunk 3]
```diff
@@ -162,6 +166,11 @@ private static int getJavascriptImportDetail(Menu menu, List<String> imports, in
             welMenu.setApplication(WELCOMEAPP);
             welMenu.setContext(WELCOMECONTEXT);
             menu.getSubMenus().put(WELCOMECONTEXT, welMenu);
+
+            Menu userJournalMenu = new Menu();
```
Comment (pbabin) [inline org.talend.mdm.webapp.general/src/main/java/org/talend/mdm/webapp/general/server/util/Utils.java]: We don't add any new menu on the navigator panel!

File: org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java [hunk 4]
```diff
@@ -250,7 +250,7 @@ private boolean isNeedToAddExplicitly(boolean isStaging, TypedExpression field)
         return isNeedToAdd;
     }
 
-    private static IWhereItem normalizeConditions(ArrayList<IWhereItem> conditions) {
+    public static IWhereItem normalizeConditions(ArrayList<IWhereItem> conditions) {
```
[the remark this answers is not in the document]
Comment (zfkang-talend) [reply org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java]: It's my fault, at the beginning I used this method to generate IwhereItem, but finally, I used SOAP API, forget to restore to private.

File: org.talend.mdm.webapp.browserecords/src/main/resources/org/talend/mdm/webapp/browserecords/client/i18n/BrowseRecordsMessages.properties (hunk 2 above)
Comment (zfkang-talend) [reply org.talend.mdm.webapp.browserecords/src/main/resources/org/talend/mdm/webapp/browserecords/client/i18n/BrowseRecordsMessages.properties]: ^M linebreak characters are only displayed on GitHub,  open one properties file on GitHub, ^M always displayed. but local ide and Git Gui, have no ^M linebreak characters.

File: org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (zfkang-talend) [reply org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java]: It's my fault, at the beginning I used this method to generate IwhereItem, but finally, I used SOAP API, forget to restore to private. now changed to private.

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/JournalAction.java [hunk 5]
```diff
@@ -61,9 +61,7 @@
 @SuppressWarnings("serial")
 public class JournalAction extends RemoteServiceServlet implements JournalService {
 
-    private static final Logger LOG = Logger.getLogger(JournalAction.class);
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/JournalAction.java]: Why do we remove this and add `protected getLogger()`, the subclass should have its own like `Logger.getLogger(SubClass.class)`.

Comment (yjli-talend) [reply org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/JournalAction.java]: we add method,and subclass override this method.
    protected Logger getLogger() {
         return Logger.getLogger(JournalAction.class);
     }

Comment (lpw82gf) [reply org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/JournalAction.java]: No, subclass could have its own `private static final Logger LOG = Logger.getLogger(SubClass.class)`, simple and easy.

File: org.talend.mdm.webapp.general/src/main/java/org/talend/mdm/webapp/general/server/util/Utils.java (hunk 3 above)
Comment (yjli-talend) [reply org.talend.mdm.webapp.general/src/main/java/org/talend/mdm/webapp/general/server/util/Utils.java]: We add a new menu 'UserJournal' just for import UserJournal gxt js. Not on the navigator panel.

File: org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java [hunk 6]
```diff
@@ -181,7 +181,7 @@
             // Order by
             String orderByFieldPath;
             OrderBy.Direction queryDirection;
-            
+
```
[the remark this answers is not in the document]
Comment (lpw82gf) [reply org.talend.mdm.core/src/com/amalto/core/delegator/IItemCtrlDelegator.java]: Please care about the logic, and for the format refactor, we have discussed too many times, you just need to ignore, other developers will review them.

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/servlet/JournalExportServlet.java [hunk 7]
```diff
@@ -117,4 +119,8 @@ private JournalSearchCriteria getCriteriaFromRequest(HttpServletRequest request)
 
         return criteria;
     }
+
+    protected JournalDBService getService() {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/servlet/JournalExportServlet.java]: Function name can be changed to `getJournalDBService`?

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/client/widget/JournalSearchPanel.java [hunk 8]
```diff
@@ -291,64 +339,81 @@ public void handleEvent(FieldEvent be) {
...
-                PostDataUtil.postData(GWT.getHostPageBaseURL() + "journal/journalExport", getCriteriaMap()); //$NON-NLS-1$
+                exportAction();
             }
         });
         this.addButton(exportButton);
     }
     
-    private ListStore<ItemBaseModel> getListStore(List<String> list) {
+    protected void searchAction() {
+        if (entityField.isValid() && sourceCombo.isValid() && startDateField.isValid() && keyField.isValid()
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/client/widget/JournalSearchPanel.java]: Could we have a single private function to check all the `isValid`?

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/client/widget/JournalSearchPanel.java [hunk 9]
```diff
@@ -438,4 +503,20 @@ public void setCurrentDataModel() {
             this.dataModelCombo.setValue(currentModel);
         }
     }
+
+    protected void exportAction() {
+        PostDataUtil.postData(GWT.getHostPageBaseURL() + "/journal/journalExport", getCriteriaMap()); //$NON-NLS-1$
+    }
+
+    protected List<String> generateOperatorList() {
+        List<String> list = new ArrayList<String>();
```
[the remark this answers is not in the document]
Comment (yjli-talend) [reply org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/client/widget/JournalSearchPanel.java]: This cause a compile error.

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/servlet/JournalExportServlet.java (hunk 7 above)
Comment (yjli-talend) [reply org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/servlet/JournalExportServlet.java]: This service need override,it is `JournalDBService` or `UserJournalDBService`.We need only one service in this class.

**Pull request merged**